Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -64,16 +122,136 @@ class PreplannedMapModel: ObservableObject, Identifiable {
         }
     }
     
-    /// A Boolean value indicating if download can be called.
-    var canDownload: Bool {
-        switch status {
-        case .notLoaded, .loading, .loadFailure, .packaging, .packageFailure,
-                .downloading, .downloaded:
-            false
-        case .packaged, .downloadFailure:
-            true
+    /// Updates the status based on the download result of the mobile map package.
+    func updateDownloadStatus(for downloadResult: Optional<Result<MobileMapPackage, any Error>>) {
+        switch downloadResult {
+        case .success:
+            status = .downloaded
+            if let mobileMapPackage = try? downloadResult?.get() {
+                self.mobileMapPackage = mobileMapPackage
+            }
```

Comment:
```suggestion
        case .success(let mobileMapPackage):
            status = .downloaded
            self.mobileMapPackage = mobileMapPackage
```
Is this functionally equivalent?